lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorControlHandler.java

```diff
@@ -69,15 +69,9 @@ public synchronized void send(final EvaluatorRuntimeProtocol.EvaluatorControlPro
       throw new IllegalStateException("Trying to send an EvaluatorControlProto before the Evaluator ID is set.");
     }
     if (!this.stateManager.isRunning()) {
-      final String msg = new StringBuilder()
-          .append("Trying to send an EvaluatorControlProto to Evaluator [")
-          .append(this.evaluatorId)
-          .append("] that is in state [")
-          .append(this.stateManager.toString())
-          .append("], not [RUNNING]. The control message was: ")
-          .append(evaluatorControlProto.toString())
-          .toString();
-      throw new IllegalStateException(msg);
+      LOG.log(Level.WARNING, "Trying to send an EvaluatorControlProto to Evaluator [{0}] that is in state [{1}], not [RUNNING]. The control message was: {2}",
+          new Object[]{this.evaluatorId, this.stateManager, evaluatorControlProto});
+      return;
```

Comment:
No, this just makes it so that we don't crash if it happens. [REEF-167](https://issues.apache.org/jira/browse/REEF-167) describes a more fundamental solution to this problem.